Documentation/design-docs/GuardedDevirtualization.md

```diff
@@ -0,0 +1,533 @@
+# Guarded Devirtualization
+
+## Overview
+
+Guarded devirtualization is a proposed new optimization for the jit in .NET Core
+3.0. This document describes the motivation, initial design sketch, and highlights
+various issues needing further investigation.
+
+## Motivation
+
+The .NET Core jit is able to do a limited amount of devirtualization for virtual
```

Comment:
Nit: "jit" is used here, in other documents "JIT" is used (if I remember correct). So I suppose to always use "JIT" to be consistent within coreclr-repo.

Reply:
Sure, will update...